### ISSUE TITLE
libssh2 official release (1.9.0 tarball is missing test files

### ISSUE THREAD

**Issue by nehaljwani** [libssh2/libssh2#379]
**Describe the bug**
libssh 1.9.0 release tarball has missing files.

**To Reproduce**
Try building official release tarball for 1.9.0 using CMake. Configure step with default option fails:
```
CMake Error at tests/CMakeLists.txt:155 (add_executable):
  Cannot find source file:

    test_public_key_auth_succeeds_with_correct_ed25519_key_from_mem.c

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .m .M .mm .h .hh .h++ .hm
  .hpp .hxx .in .txx
```
**Expected behavior**
Source tarball should build with CMake.

**Version (please complete the following information):**
- All platforms, v1.9.0

**Comment by willco007** [comment]
@bagder any ideas why this would happen?

**Comment by willco007** [comment]
Quick follow-up, the tarball is missing the new tests added post 1.8 release. Looking at the script it is not immediately obvious why it is missing those files.

**Comment by willco007** [comment]
After more investigation it looks like the script is using automake to loop in the files for the tarball, but perhaps libssh2 was built against OpenSSL 1.0 which then would ignore the 1.1 specific test files leaving them out of the release.

**Comment by bagder** [comment]
Each libssh2 release is made with 'maketgz' which is a wrapper around 'make dist' and it should of course work independently of what OpenSSL version is or isn't used. This indicates we should add a CI test that verifies that a maketgz can build fine. Also, maketgz can be tested sort of by building a [daily snapshot](https://libssh2.org/snapshots/).

**Comment by seba30** [comment]
Can you please update the official tarball, so that includes the 3 missing test files?

**Comment by bagder** [comment]
I don't think changing an already released tarball is a good idea. I'm all for a follow-up release though. I personally then prefer a "proper" release off the main branch instead of a non-tested version branched off 1.9.0.

Also, is this issue actually fixed? I haven't seen any fix/confirmation. Are the previously missing files present in the daily snapshots?

**Comment by willco007** [comment]
I haven't fixed it yet. I was going to wait and land #243 which would fix this issue. I just haven't had time to fully review it yet.

**Comment by tiennou** [comment]
I'm not sure #243 would actually fix that, and it might even make it worse. AFAIU the issue is that the official packaging is done via `maketgz`/`autotools`, which doesn't match what CMake expects ? I'll investigate what's going on but again, I'm far worse at autotools, and I'm still wondering if all this would be fixed by dropping it ?

**Comment by bagder** [comment]
How is this even an autotool issue? The problem is that the correct files to include in the distro was not listed in the correct file. That problem could appear with *any* method as long as we don't have CI builds that actually verify the release-build script.

**Comment by tiennou** [comment]
Sorry, yeah, the auto-thing that makes `make dist` work. Right now Travis has a CMake package step, nothing tests `make dist`, apart from `maketgz` which doesn't run as-is on macOS (no `sed -e` support). `tests/Makefile.am` has a TODO about those missing files an OpenSSL version check. Merging in #243 would make it worse because I haven't touched any Makefiles, so the CMake build system packaged by `make dist` would be missing even more parts.

**Comment by bagder** [comment]
> nothing tests make dist, apart from maketgz

Exactly, that's a missing CI build that we should add. Make a tarball with maketgz and then make a build from that tarball only.

> which doesn't run as-is on macOS

Surely that can be fixed if anyone thinks it is necessary.

**Comment by willco007** [comment]
The missing files no longer exist with the clar test system in #243, which is why this issue would become moot. However, the test make files need to be fixed before it can be landed.

**Comment by willco007** [comment]
Should be resolved with ecd6a74.
